### ISSUE TITLE
Cannot declare keys on a `TypedDict` with the same name as methods of `_TypedDict`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2301]
**Describe the bug**
With `reportIncompatibleMethodOverride` set to `"error"` (like it is in strict mode), users cannot define any key values which have the same name as the methods defined on `_TypedDict` from typeshed.

**To Reproduce**
Set `reportIncompatibleMethodOverride` to `"error"` and type check the following code:

```python
class MyObject(TypedDict):
    values: list[str]  # error: "values" overrides method of same name in class "_TypedDict" with incompatible type "list[str]"
```

**Expected behavior**
No error for keys named `copy`, `setdefault`, `pop`, `update`, `__delitem__`, `items`, `keys`, `values`, `__or__`, or `__ior__`.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.168

**Comment by erictraut** [comment]
Thanks for the report. It makes sense to special-case `TypedDict` and suppress this diagnostic check.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.169. It will also be included in the next release of pylance.
